### PR TITLE
Allow versioning in downloaded installer file names

### LINKED ISSUE #5271: Download Installer - Need Improvement for Naming Installer Example "PackageName_Version"
### Please confirm these before moving forward.

- [x] I have searched for my feature proposal and have not found a work-in-progress/duplicate/resolved/discarded issue.
- [x] This improvement refers to an existing feature. If you want to suggest a new feature, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=new-feature&projects=&template=feature-request.yml&title=%5BFEATURE+REQUEST%5D+%28Enter+your+description+here%29).
- [x] This improvement is not a bug. If you want to report a bug, please use [this template](https://github.com/Devolutions/UniGetUI/issues/new?labels=bug&projects=&template=bug-issue.yml&title=%5BBUG%5D+%28Enter+your+description+here%29).

### Describe the improvement

Hello,

I am working in a area that don't have proper internet connection. most of the time I download with mobile Internet which is limited. 

When i download offline installer it saves the file by Different names.

The Problems i have 

1. Sorting by name (Example "setup-lightshot-5.5.0.7", the word setup attached at front causing sort by name to fail. it takes time to find package, when there are lots of package in the folder.)

2. Not able to identify by version Like some installers don't have version attached to the file name if some newer version is available on uniget it gets difficult to verify the offline installer version as the file name does not have the version Like "SpotifyFullSetupX64", "BraveBrowserStandaloneSilentSetup", "googlechromestandaloneenterprise64" etc.
 
3. Older version gets overwritten (related to point 2. when the file name does not have version it gets overwrite by newer installer.)

If possible, make this simple change so that every version gets proper name and newer installer does not overwrites older version. 

Thanks for doing good work. Hope to implement this simple and good feature for Download Installer function.

### Describe how this improvement could help users

It will help user to identify the download installer versions. and it keeps version distinct and don't overwrite previous version those who download offline installers .it will be great help.

### PR DESCRIPTION
**Pull request by GabrielDuf**
This pull request introduces a user-configurable setting for naming downloaded installer files, allowing users to choose between several naming schemes (e.g., including version or package ID in the file name). It also implements the logic for building installer file names according to the selected scheme, updates the UI and settings infrastructure to support this option, and adds comprehensive tests for the new functionality. Additionally, it introduces a new property to package managers to indicate if the installer URL follows the package version.

**Installer file naming customization:**

* Added `InstallerFileNaming` utility with logic to build installer file names according to a user-selected scheme (publisher name, package name and version, package ID and version, or publisher name and version), including extension handling and sanitization.
* Added new setting `InstallerFileNameScheme` to the settings engine, exposed in the UI as a combobox in the Operations settings page, and indexed for search.

**Installer download flow improvements:**

* Updated the download installer action to use the new naming logic and ask the user for the save location with the generated file name. 

**Testing and reliability:**

* Added a comprehensive test suite for `InstallerFileNaming`, covering all naming schemes, extension extraction, version normalization, and edge cases.

**Package manager interface update:**

* Added `InstallerUrlFollowsPackageVersion` property to `IPackageManager` and implemented it in relevant package managers, allowing for future logic that depends on whether the installer URL changes with the package version.

### CODE REVIEW

File: src/UniGetUI.Core.Tools/InstallerFileNaming.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    {
+        ["exe"] = ".exe",
+        ["burn"] = ".exe",
+        ["inno"] = ".exe",
+        ["nullsoft"] = ".exe",
+        ["wix"] = ".exe",
+        ["portable"] = ".exe",
+        ["msi"] = ".msi",
+        ["msix"] = ".msix",
+        ["appx"] = ".msix",
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools/InstallerFileNaming.cs]: Fixed in [37c8697](https://github.com/Devolutions/UniGetUI/pull/5319/commits/37c86977f5f07c51b010bedac76482160d36c032)

File: src/UniGetUI.Avalonia/Views/Pages/SettingsPages/Operations.axaml.cs [hunk 2]
```diff
@@ -29,6 +29,24 @@ public Operations()
             ParallelOperationCount.AddItem(v, v, false);
         ParallelOperationCount.ShowAddedItems();
 
+        InstallerNameSchemeCard.AddItem(
+            "Name given by the publisher",
+            InstallerFileNaming.PublisherNameValue
+        );
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Avalonia/Views/Pages/SettingsPages/Operations.axaml.cs:35]: Fixed in [37c8697](https://github.com/Devolutions/UniGetUI/pull/5319/commits/37c86977f5f07c51b010bedac76482160d36c032)

File: src/UniGetUI.Avalonia/Views/Pages/SettingsPages/Operations.axaml [hunk 3]
```diff
@@ -41,6 +41,18 @@
                                    WarningText="{t:Translate You may lose unsaved data}"
                                    CornerRadius="8"/>
 
+            <settings:TranslatedTextBlock Text="Installer downloads"
+                                          FontSize="14"
+                                          FontWeight="SemiBold"
+                                          Margin="44,32,4,8"
+                                          automation:AutomationProperties.HeadingLevel="2"/>
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Avalonia/Views/Pages/SettingsPages/Operations.axaml:48]: Fixed in [37c8697](https://github.com/Devolutions/UniGetUI/pull/5319/commits/37c86977f5f07c51b010bedac76482160d36c032)

File: src/UniGetUI.Avalonia/Views/DialogPages/PackageDetailsWindow.axaml.cs [hunk 4]
```diff
@@ -466,10 +466,10 @@ private void DownloadInstaller()
...
         if (!_vm.Package.Manager.Capabilities.CanDownloadInstaller) return;
         Close();
-        // Hook into the platform's download path if/when wired up. For now, fall back to opening
-        // the installer URL so the user can still grab the file.
-        if (_vm.InstallerUrl is not null)
-            OpenUrl(_vm.InstallerUrl.ToString());
+        _ = AvaloniaPackageOperationHelper.AskLocationAndDownloadAsync(
+            _vm.Package,
+            TEL_InstallReferral.DIRECT_SEARCH
+        );
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Avalonia/Views/DialogPages/PackageDetailsWindow.axaml.cs]: [966c17f](https://github.com/Devolutions/UniGetUI/pull/5319/commits/966c17f1abef563dfed85970a6b276c0190a87e3)

File: src/UniGetUI.PackageEngine.PackageManagerClasses/Packages/Package.cs [hunk 5]
```diff
@@ -333,19 +333,55 @@ public virtual bool NewerVersionIsInstalled()
             return false;
         }
 
+        private string ResolveInstallerVersion()
+        {
+            if (Manager.InstallerUrlFollowsPackageVersion)
+                return VersionString;
+
+            if (IsUpgradable)
+                return NewVersionString;
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.PackageEngine.PackageManagerClasses/Packages/Package.cs:342]: [966c17f](https://github.com/Devolutions/UniGetUI/pull/5319/commits/966c17f1abef563dfed85970a6b276c0190a87e3)

File: src/UniGetUI.Avalonia/Views/DialogPages/PackageDetailsWindow.axaml.cs [hunk 6]
```diff
@@ -36,11 +36,16 @@ private enum LayoutMode { Unset, Normal, Wide }
...
     private readonly PackageDetailsViewModel _vm;
+    private readonly TEL_InstallReferral _referral;
     private InstallOptionsViewModel? _installVm;
     private InstallOptions? _installOpts;
 
-    public PackageDetailsWindow(IPackage package, OperationType operation)
+    public PackageDetailsWindow(
+        IPackage package,
+        OperationType operation,
+        TEL_InstallReferral referral = TEL_InstallReferral.DIRECT_SEARCH)
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Avalonia/Views/DialogPages/PackageDetailsWindow.axaml.cs]: [82d8017](https://github.com/Devolutions/UniGetUI/pull/5319/commits/82d801716f64288dee0b1bd04845355550d75591)

File: src/UniGetUI.Core.Tools.Tests/InstallerFileNamingTests.cs [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+namespace UniGetUI.Core.Tools.Tests;
+
+public class InstallerFileNamingTests : IDisposable
+{
+    public void Dispose()
+    {
+        Settings.SetValue(Settings.K.InstallerFileNameScheme, "");
+        GC.SuppressFinalize(this);
+    }
```
[the remark this answers is not in the document]
Comment (GabrielDuf) [reply src/UniGetUI.Core.Tools.Tests/InstallerFileNamingTests.cs:29]: [65b21fd](https://github.com/Devolutions/UniGetUI/pull/5319/commits/65b21fd9cba416fd8f18adef0be4e200115bccc5)
